ydb/core/blob_depot/agent/storage_get_block.cpp

```diff
@@ -6,15 +6,52 @@ namespace NKikimr::NBlobDepot {
     template<>
     TBlobDepotAgent::TQuery *TBlobDepotAgent::CreateQuery<TEvBlobStorage::EvGetBlock>(std::unique_ptr<IEventHandle> ev) {
         class TGetBlockQuery : public TBlobStorageQuery<TEvBlobStorage::TEvGetBlock> {
+            ui32 BlockedGeneration = 0;
+            ui32 RetriesRemain = 10;
+
         public:
             using TBlobStorageQuery::TBlobStorageQuery;
 
             void Initiate() override {
-                // TODO: implement GetBlock logic for BlobDepot
+                const auto status = Agent.BlocksManager.CheckBlockForTablet(Request.TabletId, 0, this, &BlockedGeneration);
+                if (status == NKikimrProto::OK) {
+                    EndWithSuccess(std::make_unique<TEvBlobStorage::TEvGetBlockResult>(NKikimrProto::OK, Request.TabletId, BlockedGeneration));
+                } else if (status != NKikimrProto::UNKNOWN) {
+                    EndWithError(status, "tablet was deleted");
```

Comment:
why? which tablet is deleted? maybe it's ERROR and BlobDepot tablet is just unreachable at the moment?